### PR TITLE
Merge `wcs-migration` into `trunk`

### PR DESCRIPTION
**Pull request by Ferdev**
## Description
This PR merges all work done in the `wcs-migration` branch to trunk, to prepare for the future release of the Migration Flow project.

### Related issue(s)
N/A

### Steps to reproduce & screenshots/GIFs
N/A

### Checklist

- [ ] unit tests

- [ ] `changelog.txt` entry added
- [ ] `readme.txt` entry added

### CODE REVIEW

**Review by harriswong — APPROVED** [review]
I pulled this feature branch and did a quick smoke test with WCShipping disabled. I can confirm the migration modal. I clicked through the modal and I was able to get 2 plugins activated
image

I can also get the new WCShipping banner to show inside an order.

**Question**
Double checking if there are any tasks we need to do before merging this? Otherwise, LGTM 👍

**Comment by Ferdev** [comment]
> Double checking if there are any tasks we need to do before merging this? Otherwise, LGTM 👍

Thanks for the question Harris! We can totally merge this now, but it won't be enabled for regular accounts until the migration flag in WooCommerce Connect Server is activated for `production` env. Other than that, I see no issues merging it now.

File: client/components/migration/feature-announcement.jsx
Comment (kallehauge) [inline client/components/migration/feature-announcement.jsx]: Very late/last minute suggestion: The "providing the functionality to carry" part made me go back and re-read it, so perhaps a simple change like this would be OK last minute? 😄 

```suggestion
					<p>{translate('We will ensure a seamless transition by transferring of all your settings and shipping labels when you update.')}</p>
```

File: client/components/migration/feature-announcement.jsx
Comment (kallehauge) [inline client/components/migration/feature-announcement.jsx:23]: I noticed that some of the padding around our "content" is showing a purple border. Should this be white?

Screenshot 2024-08-20 at 16 55 02

File: changelog.txt
Comment (kallehauge) [inline changelog.txt]: I read this as the functionality already existed since it's a UI comment - can be ignored 😄 

```suggestion
* Add - A new migration experience from this plugin to the newly released WooCommerce Shipping plugin.
```

**Review by kallehauge — APPROVED** [review]
Works as expected 👏  

I gave this a spin to test the whole flow while reviewing https://github.com/woocommerce/woocommerce-shipping/pull/667 and found a few minor things. None of them are blocking in any way!

_(Edit: I've not reviewed the code itself - purely communicating I had a successful experience from a manual testing point of view 😄)_

File: client/components/migration/feature-announcement.jsx
Comment (harriswong) [reply client/components/migration/feature-announcement.jsx:23]: Yes there shouldn't be a gap 0EcJhbcadV2uAqVrA8hb6a-fi-467_5757. I created a follow up for this https://github.com/Automattic/woocommerce-services/issues/2801.

File: changelog.txt
Comment (harriswong) [reply changelog.txt]: Looks good to me, I am committing your suggestion.

File: client/components/migration/feature-announcement.jsx
Comment (harriswong) [reply client/components/migration/feature-announcement.jsx]: Looks good to me, i am committing your suggestion.

Comment (harriswong) [reply client/components/migration/feature-announcement.jsx]: Actually, I will change it to "We will ensure a seamless transition by transferring ~~of~~ all your settings and shipping labels when you update."

Comment (harriswong) [reply client/components/migration/feature-announcement.jsx]: Fixed in 71637e12d01a01cfeb932deebcfba170472d3e28

**Comment by harriswong** [comment]
👋 Hi @kallehauge! I addressed the comments, please take a look again!

File: client/components/migration/feature-announcement.jsx
Comment (kallehauge) [reply client/components/migration/feature-announcement.jsx:23]: Thank you Harris! IMO it's not blocking and honestly could look like it's intentional, so I'm marking this comment as resolved ❤️ 

(CC: @kloon, courtesy ping since you're the DRI)

**Review by kallehauge — APPROVED** [review]
I'm a happy man, @harriswong! 😄 

Thank you for looking into my comments 🙇

**Pull request merged**
